src/hssm/distribution_utils/dist.py

```diff
@@ -356,8 +357,210 @@ def rng_fn(
     return SSMRandomVariable
 
 
+def make_custom_rv(
```

Comment:
Does this function have much in common with the previous one? I wonder if we can somehow combine these two into a more general function so we don't repeat the same code?

Also, seems like codecov is complaining that this function was not tested